src/Libraries/Watch3DNodeModelsWpf/HelixWatch3DNodeViewModel.cs

```diff
@@ -86,8 +128,7 @@ protected override void PortDisconnectedHandler(PortModel obj)
             var updatedNode = dynamoModel.CurrentWorkspace.Nodes.FirstOrDefault(n => n.GUID == node.GUID);
             if (updatedNode == null) return;
 
-            var visibleUpstream = new List<NodeModel>();
-            watchModel.VisibleUpstreamNodes(visibleUpstream);
+            var visibleUpstream = watchModel.InPorts.SelectMany(p => p.Connectors.Select(c => c.Start.Owner)).ToList();
```

Comment:
why use `ToList()` here ?

Reply:
also shouldn't this variable just be called `upstream` ?

Reply:
Yep, yep